File: src/main/java/com/exampledigisphere/vitapetcare/admin/permission/Permission.java
package com.exampledigisphere.vitapetcare.admin.permission;

import com.exampledigisphere.vitapetcare.config.root.BaseEntity;
import com.fasterxml.jackson.annotation.JsonView;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "per_permissions")
public class Permission extends BaseEntity {

  public interface Json {
    interface List {
    }

    interface Detail extends List {
    }

    interface All extends Detail {
    }
  }

  @NotBlank
  @JsonView(Json.List.class)
  private String name;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }
}

File: src/main/kotlin/com/exampledigisphere/vitapetcare/permission/domain/Permission.kt
package com.exampledigisphere.vitapetcare.permission.domain

import com.exampledigisphere.vitapetcare.config.root.BaseEntity
import com.exampledigisphere.vitapetcare.roles.domain.Role
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.ManyToMany
import jakarta.persistence.Table

@Entity
@Table(name = "per_permissions")
class Permission : BaseEntity() {
  lateinit var name: String;

  @ManyToMany(mappedBy = "permissions", fetch = FetchType.LAZY)
  val roles: MutableSet<Role> = HashSet()
}
